torch/_jit_internal.py

```diff
@@ -724,7 +724,11 @@ def _qualified_name(obj):
     if isinstance(obj, torch._C.ScriptFunction):
         return obj.qualified_name
 
-    name = obj.__name__
+    if getattr(obj, "__name__", None):
```

Comment:
Why did we need to make this change ? Seems a little strange

Reply:
Weird. nit: maybe `elif isinstance(obj, Enum)` in elif conditional instead so it's a little more clear why this is needed

Reply:
Indeed very weird. Done.